core/lib/contract_verifier/src/etherscan/client.rs

```diff
@@ -0,0 +1,329 @@
+}
+
+fn process_api_call_result(
+    result: Result<impl AsRef<str>, reqwest::Error>,
+) -> Result<String, ApiError> {
+    match result {
+        Ok(result) => {
+            let result = result.as_ref();
+            let raw_response: ApiResponseRaw = serde_json::from_str(result).map_err(|error| {
+                if result.contains("Page not found") {
+                    ApiError::PageNotFound
+                } else if is_blocked_by_cloudflare_response(result) {
+                    ApiError::BlockedByCloudflare
+                } else if is_cloudflare_security_challenge(result) {
+                    ApiError::CloudFlareSecurityChallenge
+                } else {
+                    ApiError::Serde {
+                        error,
+                        content: result.to_string(),
+                    }
+                }
+            })?;
+
+            match raw_response.status.as_str() {
+                "0" => {
+                    if raw_response.result == "Contract source code not verified" {
+                        return Err(ApiError::ContractNotVerified);
+                    }
+                    if raw_response.result == "Contract source code already verified" {
+                        return Err(ApiError::ContractAlreadyVerified);
+                    }
+                    if raw_response.result == "Pending in queue" {
+                        return Err(ApiError::VerificationPending);
+                    }
+                    // There is a number of daily limit in between the checked values.
+                    // I don't want to rely on the exact number as it is a subject to change.
+                    if raw_response.result.starts_with("Daily limit")
+                        && raw_response
+                            .result
+                            .ends_with("source code submissions reached")
+                    {
+                        return Err(ApiError::DailyVerificationRequestsLimitExceeded);
+                    }
+                    let result_lower = raw_response.result.to_lowercase();
+                    if result_lower.contains("rate limit reached") {
+                        return Err(ApiError::RateLimitExceeded);
+                    }
+                    // Error message can be "Invalid API Key" or "Missing/Invalid API Key"
+                    if result_lower.contains("invalid api key") {
+                        return Err(ApiError::InvalidApiKey);
+                    }
+                    Err(ApiError::ErrorResponse {
+                        message: raw_response.message,
+                        result: raw_response.result,
+                    })
+                }
+                "1" => Ok(raw_response.result),
+                _ => Err(ApiError::UnexpectedResponse {
+                    message: raw_response.message,
+                    result: raw_response.result,
+                }),
+            }
+        }
+        Err(err) => Err(ApiError::Reqwest(err)),
```

Comment:
`ApiError` can already has `#[from] reqwest::Error`, so I guess you can convert it automatically